File: WebApplication/src/app/order/order.ts

export class Order {
  public _id: string;
  public orderCreationDate: string;
  constructor(
    public customerName: string,
    public customerEmail: string,
    public customerAddress: string,
    public cartTotalPrice: number,
    public cartItems: { productId: string; productQuantity: number }[]
  ) {}
}

File: src/main/java/com/BakeryApplication/Order/Order.java
package com.BakeryApplication.Order;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDate;

@Document(collection = "Order")
@Data
public class Order {
    @Id
    String _id;
    String customerName;
    String customerEmail;
    String customerAddress;
    String orderCreationDate;
    double cartTotalPrice;
    CartItem[] cartItems;
}
